### PR TITLE
🔧 Standardise CI

### PR DESCRIPTION
**Pull request by chrisjsewell**
See https://github.com/marvel-nccr/quantum-mobile/issues/132

### CODE REVIEW

**Comment by chrisjsewell** [comment]
@adegomme as mentioned above, I'm moving all the role repos to a "standardised structure". This includes moving to GH Actions and using different base docker images for the testing.
As you can see in the checks, this is failing for Ubuntu 1804 😢 (also for 2004, but that doesn't need to be addressed right now)
Can you have a look and see if there is any obvious fix?

(note, it might just be that the image lacks some packages, I'm making a PR like this for all codes, so we shall see if they also have this issue)

**Comment by adegomme** [comment]
I'm trying to reproduce locally, as molecule seems to cut the output we see before the failure (it squashes everything into one line, which has limits on linux, and then seems to remove all further stderr output)... 
I had some failures as I had ansible-galaxy setup with an old version of the role and couldn't find out why it was using an old bigdft for a while ... Now I'm looking at how to get the complete log.

**Comment by adegomme** [comment]
with --debug I was able to see that it's complaining about not finding wget (or curl).  I'm trying to add wget (and bzip2 as we use a tar.bz at this step) to the installed packages requirements to see if it's enough (they are already on quantum-mobile, so that was not apparent before ).

**Comment by chrisjsewell** [comment]
thanks, yeh for reference, this is the old docker build:

https://github.com/marvel-nccr/ubuntu-docker-base/blob/master/Dockerfile

and this is the new one and what is derived from

https://github.com/marvel-nccr/docker-ubuntu1804-ansible/blob/master/Dockerfile
https://github.com/geerlingguy/docker-ubuntu1804-ansible/blob/master/Dockerfile

bit of a pain to change, but hopefully better in the long run

**Comment by adegomme** [comment]
I just passed it locally, so let's see.

File: defaults/main.yml [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
-bigdft_version: "1.9.0"
-bigdft_src: bigdft-suite-{{ bigdft_version }}
-bigdft_src_archive: "{{ bigdft_src }}.tar.gz"
+bigdft_version: "4855f2cd" # "1.9.0"
 bigdft_code_folder: "/tmp"
 bigdft_prefix: "/usr/local"
 bigdft_python_path: "{{ ansible_python_interpreter | default('/usr/bin/python') }}"
-bigdft_url: "https://gitlab.com/l_sim/bigdft-suite/-/archive/{{ bigdft_version }}/{{ bigdft_src_archive }}"
 bigdft_topdir: "{{ bigdft_code_folder }}/bigdft-suite"
+# bigdft_src: bigdft-suite-{{ bigdft_version }}
```
Comment (chrisjsewell) [inline defaults/main.yml:6]: FYI @adegomme I also commented all these out, since we are not actually using them

**Comment by adegomme** [comment]
20.04 still fails, apparently because of libblacs-mpi-dev, which is no longer installable, but provided by libscalapack-mpi-dev. Fixing this as well.

File: defaults/main.yml (hunk 1 above)
Comment (adegomme) [reply defaults/main.yml:6]: yes, it's still temporary while we use a git version, so I will activate it back when needed.

**Comment by chrisjsewell** [comment]
> 20.04 still fails, apparently because of libblacs-mpi-dev, which is no longer installable, but provided by libscalapack-mpi-dev. Fixing this as well.

I'd maybe not do this now and just remove the `ubuntu2004`

**Comment by chrisjsewell** [comment]
Its not necessary to support 20.04 just yet, although obviously is it is easy to support then we should. If that requires extra packages to install then maybe not worth it yet? or make the install only run if on 20.04

**Comment by chrisjsewell** [comment]
hmm, are we encountering the hanging install issue?

**Comment by adegomme** [comment]
> hmm, are we encountering the hanging install issue?

no, it takes a while to build, but no hanging with this version (I tried this weekend on a fork btw : 
https://github.com/adegomme/quantum-mobile/actions/runs/312812103 )

**Comment by adegomme** [comment]
> Its not necessary to support 20.04 just yet, although obviously is it is easy to support then we should. If that requires extra packages to install then maybe not worth it yet? or make the install only run if on 20.04

Well it's actually a package to remove. I need to check if 16.04 still passes without it, though.
20.04 complained it was missing a C++ compiler, btw, I had to add g++ as a dependency, it's building now. Maybe it's missing from the docker ?

**Comment by chrisjsewell** [comment]
I'm going to remove ubuntu2004 for now and we sort that in a seperate PR if thats ok?

**Comment by adegomme** [comment]
no, it works :p
just waiting for 16.04 to completely finish building, but it seems to be heading to a success.

File: README.md [hunk 2]
```diff
@@ -14,14 +17,58 @@ See `defaults/main.yml`
...
+```bash
+pip install pre-commit
+pre-commit run -all
+```
+
+## Deployment
+
+Deployment to Ansible Galaxy is automated *via* GitHub Actions.
+Simply tag a release `vX.Y.Z` to initiate the CI and release workflow.
+Note, the release will only complete if the CI tests pass.
```
Comment (chrisjsewell) [inline README.md:65]: sweet! Once it passes, I'll release a new version using the new deployment automation 😄

Comment (adegomme) [reply README.md:65]: it just finished locally .. Do I put 20.04 back in this branch ?

Comment (chrisjsewell) [reply README.md:65]: nah lets handle that in a separate PR (just open it once this is merged). I've got no patience left to wait for another build 😆

Comment (chrisjsewell) [reply README.md:65]: then I can get it in to quantum mobile (which is 18.04 anyway) and start that build

**Pull request merged**